clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -166,6 +166,9 @@ static LogicalResult checkConstantTypes(mlir::Operation *op, mlir::Type opType,
     return success();
   }
 
+  if (mlir::isa<cir::ArrayType>(opType))
```

Comment:
I don't see this in the incubator implementation of this function. Why is this here?